stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1619,10 +1617,12 @@ function reflect!(x::AbstractVector, y::AbstractVector, c, s)
     if n != length(y)
         throw(DimensionMismatch(lazy"x has length $(length(x)), but y has length $(length(y))"))
     end
-    @inbounds for i = 1:n
-        xi, yi = x[i], y[i]
-        x[i] =      c *xi + s*yi
-        y[i] = conj(s)*xi - c*yi
+    for i = eachindex(x,y)
+        @inbounds begin
```

Comment:
Hm, that seems a bit weird to add another `begin ... end` group. I'd suggest to leave the annotation in front of the `for`-loop, since it's just as tight in scope as this one.

Reply:
The difference probably doesn't matter in this case as we use linear indexing, but for Cartesian indexing, I've recently found that putting the `@inbounds` on the loop can slow down the iterations (likely because it taints effects).
https://github.com/JuliaLang/julia/pull/55919